### PR TITLE
Extended indexing and CacheRefresher notifications integration

### PR DESCRIPTION
**Pull request by acoumb**
Current PR contains a series of updates to the indexing service based on these reported issues: #92 and #90 , as well as an integration with [Cache Refresher](https://docs.umbraco.com/umbraco-cms/reference/notifications/cacherefresher-notifications).

Also I have identified a bug with the payload sent to Algolia when a variant of a content node got Unpublished, due to the use of `AvailableCultures` instead of `PublishedCultures` in the record content builder.

### CODE REVIEW

File: src/Umbraco.Cms.Integrations.Search.Algolia/Handlers/AlgoliaContentCacheRefresherHandler.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            foreach (ContentCacheRefresher.JsonPayload payload in payloads)
+            {
+                if (payload.ChangeTypes != TreeChangeTypes.RefreshNode 
+                    && payload.ChangeTypes != TreeChangeTypes.RefreshBranch)
+                {
+                    return;
+                }
+
+                var x = _contentService.GetById(1205);
+                var y = x.Published;
```
[the remark this answers is not in the document]
Comment (acoumb) [reply src/Umbraco.Cms.Integrations.Search.Algolia/Handlers/AlgoliaContentCacheRefresherHandler.cs]: Test code removed.

File: src/Umbraco.Cms.Integrations.Search.Algolia/Handlers/AlgoliaContentCacheRefresherHandler.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+using Umbraco.Cms.Core.Services.Changes;
+using Umbraco.Cms.Core.Services;
+using Umbraco.Cms.Core.Routing;
+using Umbraco.Cms.Integrations.Search.Algolia.Migrations;
+using Umbraco.Cms.Integrations.Search.Algolia.Services;
+using Umbraco.Cms.Core.Models.PublishedContent;
+
+namespace Umbraco.Cms.Integrations.Search.Algolia.Handlers
+{
+    public class AlgoliaContentCacheRefresherHandler : BaseContentHandler, INotificationAsyncHandler<ContentCacheRefresherNotification>
```
[the remark this answers is not in the document]
Comment (acoumb) [reply src/Umbraco.Cms.Integrations.Search.Algolia/Handlers/AlgoliaContentCacheRefresherHandler.cs]: All use cases are covered: publish, unpublish and delete.

File: src/Umbraco.Cms.Integrations.Search.Algolia/Handlers/AlgoliaContentCacheRefresherHandler.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+        public async Task HandleAsync(ContentCacheRefresherNotification notification, CancellationToken cancellationToken)
+        {
+            if (notification.MessageObject is not ContentCacheRefresher.JsonPayload[] payloads)
+            {
+                return;
+            }
+
+            var refreshedContent = _contentService
```
[the remark this answers is not in the document]
Comment (acoumb) [reply src/Umbraco.Cms.Integrations.Search.Algolia/Handlers/AlgoliaContentCacheRefresherHandler.cs:78]: Updated, no impact on the use case.

File: src/Umbraco.Cms.Integrations.Search.Algolia/Handlers/AlgoliaContentCacheRefresherHandler.cs [hunk 4]
```diff
@@ -36,6 +61,20 @@ public async Task HandleAsync(ContentCacheRefresherNotification notification, Ca
                 return;
             }
 
+            switch (_serverRoleAccessor.CurrentServerRole)
+            {
+                case ServerRole.Subscriber:
+                    _logger.LogDebug("Umbraco Forms scheduled record deletion task will not run on subscriber servers.");
```
Comment (AndyButland) [inline src/Umbraco.Cms.Integrations.Search.Algolia/Handlers/AlgoliaContentCacheRefresherHandler.cs:67]: These log messages should be updated.

**Review by AndyButland — APPROVED** [review]
Other than the one comment, looks good to me.

**Pull request merged**